### ISSUE TITLE
loss

### ISSUE THREAD

**Issue by hxqiu** [jingyaogong/minimind-v#32]
model.py:446行计算了一次loss，单并没有使用，外边又对model(x,y).logits做了一次loss，是重复计算吗，还是有什么原因呢
![image](https://github.com/user-attachments/assets/f5a01e26-dfa0-4f99-b451-0849777ac22b)

![image](https://github.com/user-attachments/assets/a799509c-3cd5-4363-8e64-57fd127ef169)

**Comment by hxqiu** [comment]
model.py:446行计算了一次loss，外边又对model(x,y).logits做了一次loss，是重复计算吗，还是有什么原因呢

**Comment by jingyaogong** [comment]
健壮性coding

一些第三方框架会取model.last_loss，不会在外面计算
